vault_core/src/vault.rs

```diff
@@ -300,22 +300,22 @@ impl Vault {
     // ------------------------------------------
 
     #[inline(always)]
-    pub fn is_update_needed(&self, slot: u64, epoch_length: u64) -> bool {
-        let last_updated_epoch = self
-            .last_full_state_update_slot
-            .checked_div(epoch_length)
-            .unwrap();
-        let current_epoch = slot.checked_div(epoch_length).unwrap();
-        last_updated_epoch < current_epoch
+    fn is_update_needed(&self, slot: u64, epoch_length: u64) -> Option<bool> {
+        let last_updated_epoch = self.last_full_state_update_slot.checked_div(epoch_length)?;
```

Comment:
i don't think we need this, checked_div only fails if denominator is 0. that should never happen

Reply:
Yeah, make sense! I will keep unwrap